Issue 840 discover details list screen

Dlewis89 opened the pull request:
# Description
Add details screen to the discover trending s
<!-- Provide a description of the changes made -->

## Checklist

<!-- Check off [x] once complete or if not applicable -->

- [x] Screenshots added for any UX changes
![Screenshot 2024-10-13 at 16-09-08 Streamability](https://github.com/user-attachments/assets/b76cd2ad-35f0-45cf-b2bf-24249c3d8b4a)
![Screenshot 2024-10-13 at 16-07-06 Streamability](https://github.com/user-attachments/assets/4570eda2-ea58-4bf4-8d23-e71297c9db31)
- [x] Demos attached for animations/interactions
- [x] Pointed at proper branch (`develop` not `main`)
- [x] Assigned PR to yourself
- [x] Requested review from code owners
- [x] Relevant labels added (`feature`, `documentation`, etc.)
- [x] `Closes #<issue-number>` added if this resolves a GitHub issue
- [x] Branch is up to date with develop, ran `git rebase develop` if necessary
- [x] All GitHub Actions are passing
- [x] `npm run todo-ci` ran if any todo comments were created

Thenlie commented:
A side effect of using the search result header here is that the text will read `Search results for: ...`. These discover screens are not actually search results, so I don't think this makes sense. I also personally prefer the centered title, but could definitely be convinced on the left align as well. The main issue is the text. We should just display "Trending" or "Newly Added".
| Before | After |
|--------|--------|
| <img width="993" alt="image" src="https://github.com/user-attachments/assets/cdf3bcde-33b8-4ac5-ac17-ed2c9fc9c7a5"> | <img width="1725" alt="image" src="https://github.com/user-attachments/assets/44df69a1-d4af-4c2a-8678-f50f9f620a50"> |

Thenlie commented on src/screens/discover/DiscoverDetailScreen.tsx:
```diff
@@ -71,34 +61,76 @@ const requestHandler = async (props: RequestHandlerProps) => {
-                ))}
-            </div>
+            <DetailScreen
+                query={`trending`}
```
This should not be hardcoded. It should be using the `PATHS` object from line 18.

Dlewis89 replied:
fix path value so it shows the title bar and name.

Thenlie commented on src/screens/discover/DiscoverDetailScreen.tsx:
```diff
@@ -71,34 +61,76 @@ const requestHandler = async (props: RequestHandlerProps) => {
+    }, [data, hash, viewState, profile]);
 
-    // TODO: Create loader #839
+    // TODO: Create loader #839r
```
```suggestion
    // TODO: Create loader #839
```

Thenlie replied:
Will need to re-run `todo-ci` after this change.

Dlewis89 replied:
corrected typo and ran  todo-ci

Thenlie commented on src/screens/discover/DiscoverDetailScreen.tsx:
```diff
@@ -71,34 +61,76 @@ const requestHandler = async (props: RequestHandlerProps) => {
+    const initialView = storageItem === 'grid' ? 'grid' : 'list';
+    const [viewState, setViewState] = useState<'list' | 'grid'>(initialView);
+    const [hash, setHash] = useState<number>(1);
+    const {
+        data,
+        setData,
+        loading: dataLoading,
+        moreToFetch,
+        refetch,
+    } = usePaginatedData({ query: query });
```
<img width="1435" alt="image" src="https://github.com/user-attachments/assets/08c12625-eeb1-49e7-9ff8-71d5b8b11c3e">
The data seems to be broken in the PR preview. I think it would simplify this PR to remove the pagination, and instead just keep the data state we had before. The pagination can be added in #838

Dlewis89 replied:
remove the pagination for the discover detail screen.

Thenlie commented on src/screens/discover/DiscoverDetailScreen.tsx:
```diff
@@ -71,34 +61,76 @@ const requestHandler = async (props: RequestHandlerProps) => {
         .slice(1);
-    const [data, setData] = useState<ShowData[] | null>(null);
     const [loading, setLoading] = useState<boolean>(true);
+    const storageItem = localStorage.getItem('streamabilityView');
```
It's worth noting that this will be shared between discover details and search results. Meaning if you toggle the view on search results, it would also update how discover details is displayed. Not entirely sure if this is a problem or not. But we could use a unique key to avoid this.

Dlewis89 replied:
separated the local storage keys for results and discover details screens.

Thenlie reviewed (changes requested):
Nice work here! Definitely a size-able change which is awesome. In general, the view toggle works well, but found some other issues. Let me know if you have any questions regarding these. Feel free to reply on the PR or directly on Discord.

Thenlie commented:
There were a couple of lint issues caught on the CI as well. You should be able to view them here: https://github.com/Thenlie/Streamability/actions/runs/11317587229/job/31471317676?pr=1062#step:6:8. Adding a screenshot as well in case you have trouble viewing that link. 
<img width="744" alt="image" src="https://github.com/user-attachments/assets/e9c1e442-8c11-4c15-9a58-a84ee0387725">

Dlewis89 commented:
Code should reflect all change request. :)

Dlewis89 reviewed (commented):
Should be ready

Thenlie commented on src/screens/search_results/SearchResultsHeader.tsx:
```diff
@@ -109,100 +126,109 @@ const SearchResultsHeader: React.FC<SearchResultsHeaderProps> = ({
             <Typ variant='h5' alignSelf='center' margin={1}>
-                Search results for: <span className='underline'>{query}</span>
+                {query && (
+                    <p>
+                        Search results for: <span className='underline'>{query}</span>
+                    </p>
```
We should use `Typography` element instead of `p`. 
We should also only prepend the "Search results for: " on the search results screen, not the discover detail.

Thenlie commented on src/screens/search_results/SearchResultsHeader.tsx:
Since this component has evolved here to become shared between multiple screens, I think it should be moved to `src/components` and renamed to "SortFilterHeader" or something like that.

Dlewis89 replied:
Im off tomorrow, i should be able to finish up these last couple things then. :)

Dlewis89 commented:
![Screenshot 2024-10-21 at 09-04-14 Streamability](https://github.com/user-attachments/assets/8f5f364e-48cd-477b-8bdb-c24987e2f8f6)
![Screenshot 2024-10-21 at 09-04-30 Streamability](https://github.com/user-attachments/assets/11053eb0-80b0-4c77-b66f-976ab1849421)

Should  be good to go.  Renamed and moved to components folder and added the Typ back instead of p tag.

Thenlie commented:
![image](https://github.com/user-attachments/assets/ef66a5d0-5c5f-45c9-8d2f-5096307a69d8)

Still getting this build error.

Dlewis89 commented:
@Thenlie Sorry, I missed that. I double checked my logs and everything seems to be good now. No errors. Hopefully we are good to go.

Thenlie commented:
> @Thenlie Sorry, I missed that. I double checked my logs and everything seems to be good now. No errors. Hopefully we are good to go.

This is still not working. The prop was added to the component but nothing was passed in. I would just remove the most recent changes and pass in a key hardcoded, rather than through props. Let me know if you would like an example.

Dlewis89 commented:
Just want to make sure i understand correctly, are you saying to hard code the state viewStateKey in the Empty Result and Result Loader components?

Dlewis89 commented:
> ,

Just want to make sure i understand correctly, are you saying to hard code the state viewStateKey in the Empty Result and Result Loader components?

Thenlie commented:
> Just want to make sure i understand correctly, are you saying to hard code the state viewStateKey in the Empty Result and Result Loader components?

Yes that's correct.

Dlewis89 commented:
> > Just want to make sure i understand correctly, are you saying to hard code the state viewStateKey in the Empty Result and Result Loader components?
> 
> Yes that's correct.

Got it, okay ill get it done. Thank you for patience and walking me through everything. It really helps me learn more and understand the project better.

Dlewis89 commented:
@Thenlie :crossed_fingers:

Thenlie commented on src/components/SortFilterHeader.tsx:
```diff
@@ -131,13 +131,15 @@ const SortFilterHeader: React.FC<SearchResultsHeaderProps> = ({
                 </Typ>
             )}
-            <div className="">
+            <div className=''>
```
```suggestion
            <div>
```

Thenlie commented on src/screens/search_results/EmptySearchResults.tsx:
`viewStateKey` needs to be removed from the prop type here. Other changes look good 👍

Thenlie reviewed (changes requested):
![image](https://github.com/user-attachments/assets/871ff6ca-adf5-479a-a052-c5d53761308e)

One more of these props that was missed. I think everything should be good to go after this change. All the updates you made look good!

Thenlie commented:
![image](https://github.com/user-attachments/assets/00fbbb32-f830-4887-a2d5-15cc85861a69)

Updated screenshot. I think this is good to go, but probably want to inline the buttons with the title in a subsequent PR.

Thenlie reviewed (approved):
I think this is all set, thanks for all the work here!
